vslam/feature/orb_feature.py

```diff
@@ -0,0 +1,212 @@
+from typing import List, Tuple, Optional
+import operator
+import numpy as np
+from .feature import FeatureExtractor
+import cv2
+import time
+
+class ORBFeatureExtractor(FeatureExtractor):
+    """
+    Feature Extractor for ORB
+    """
+    def __init__(self, **kwargs) -> None:
+        """Contructor
+            Args: 
+        """
+        super().__init__()
+        self.orb = cv2.ORB_create(**kwargs)
+
+        self.nlevels = self.orb.getNLevels()
+        self.scaleFactor = self.orb.getScaleFactor()
+        self.edgeThreshold = self.orb.getEdgeThreshold()
+        self.fastThreshold = self.orb.getFastThreshold()
+        self.maxFeatures = self.orb.getMaxFeatures()
+        self.pathSize = self.orb.getPatchSize()
+        self.fast = cv2.FastFeatureDetector_create()
+
+        factor = 1. / self.scaleFactor
+        nDesiredFeaturesPerScale = self.maxFeatures * (1 - factor) / (1. - np.power(factor, self.nlevels))
+        self.featurePerLevel = np.zeros((self.nlevels))
+        sumFeatures = 0
+        for i in range(self.nlevels - 1): # baffled at why nlevel - 1
+            self.featurePerLevel[i] = np.round(nDesiredFeaturesPerScale).astype(int)
+            sumFeatures += self.featurePerLevel[i]
+            nDesiredFeaturesPerScale *= factor
+        self.featurePerLevel[self.nlevels - 1] = np.max([self.maxFeatures - sumFeatures, 0]).astype(int)
+
+        self.mvScaleFactor = np.arange(self.nlevels)
+        self.mvScaleFactor = np.power(self.scaleFactor, self.mvScaleFactor)
+        self.PATCH_SIZE = 31
+
+    def detect(self, img: np.ndarray, mask: np.ndarray = None) -> list:
+        """Wrapper class for ORB detector
+            Args:
+                Image
+                Mask
+
+            Returns:
+                List of keypoints
+        """
+        img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        return self.orb.detect(img, mask=mask)
+
+    def compute(self, img: np.ndarray, kp: list) -> Tuple[list, np.ndarray]:
+        """Wrapper class for ORB compute
+            Args:
+                Image
+
+            Returns:
+                Tuple of keypoint and corresponding feature descriptor
+        """
+        img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        return self.orb.compute(img, kp)
```

Comment:
shouldn't self._computeKeypoints be called here, if you are implementing it differently?

Reply:
Wait sorry I'm not done changing this yet 😬 I had to change to a different machine. I'm changing this back to simply grid-ing the image and do detect from there.